dartagnan/src/main/java/com/dat3m/dartagnan/program/event/arch/lisa/RMW.java

```diff
@@ -40,10 +40,10 @@ public ExprInterface getMemValue(){
         return value;
     }
 
-	@Override
-	public ImmutableSet<Register> getDataRegs() {
-		return value.getRegs();
-	}
+    @Override
+    public Set<Register.Read> getRegisterReads() {
+        return Register.collectRegisterReads(value, Register.UsageType.DATA, super.getRegisterReads());
```

Comment:
The address dependencies are collected in the last parameter by calling `super.getRegisterReads()`, right?

Reply:
Yes. I did it like this so can you either create a new set with the `RegisterReads` or collect them into an existing set.